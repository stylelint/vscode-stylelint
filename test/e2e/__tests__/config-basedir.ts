import { assertDiagnostics, waitForDiagnostics, openDocument, closeAllEditors } from '../helpers';

describe('"stylelint.configBasedir" setting', () => {
	afterEach(async () => {
		await closeAllEditors();
	});

	it('should resolve referenced configs using the base directory', async () => {
		const document = await openDocument('config/config-basedir.css');
		const diagnostics = await waitForDiagnostics(document);

		assertDiagnostics(diagnostics, [
			{
				code: 'color-hex-alpha',
				codeDescription: 'https://stylelint.io/user-guide/rules/color-hex-alpha',
				message: 'Expected alpha channel in "#fff" (color-hex-alpha)',
				range: [1, 9, 1, 13],
				severity: 'error',
			},
			{
				code: 'color-hex-length',
				codeDescription: 'https://stylelint.io/user-guide/rules/color-hex-length',
				message: 'Expected "#fff" to be "#ffffff" (color-hex-length)',
				range: [1, 9, 1, 13],
				severity: 'error',
			},
		]);
	});
});
