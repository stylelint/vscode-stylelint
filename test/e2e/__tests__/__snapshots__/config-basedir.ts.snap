// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"stylelint.configBasedir" setting should resolve referenced configs using the base directory 1`] = `
[
  {
    "code": "color-hex-alpha",
    "codeDescription": {
      "href": "https://stylelint.io/user-guide/rules/color-hex-alpha",
    },
    "message": "Expected alpha channel in "#fff" (color-hex-alpha)",
    "range": {
      "end": {
        "character": 13,
        "line": 1,
      },
      "start": {
        "character": 9,
        "line": 1,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
  {
    "code": "color-hex-length",
    "codeDescription": {
      "href": "https://stylelint.io/user-guide/rules/color-hex-length",
    },
    "message": "Expected "#fff" to be "#ffffff" (color-hex-length)",
    "range": {
      "end": {
        "character": 13,
        "line": 1,
      },
      "start": {
        "character": 9,
        "line": 1,
      },
    },
    "severity": 1,
    "source": "Stylelint",
  },
]
`;
